src/shared/components/home/login.tsx

```diff
@@ -107,6 +115,46 @@ async function handleLoginSubmit(i: Login, event: any) {
   }
 }
 
+export async function handleUseOAuthProvider(d: {
+  i: Login | undefined;
```

Comment:
You should be able to use 

`this.handleUseOauthProvider = this.handleUseOauthProvider.bind(this);`

Then you don't need to pass in the login.